### ISSUE TITLE
Add support for debian overrides

### ISSUE THREAD

**Issue by cpuguy83** [Azure/dalec#427]
Following up on #396 

debbuild has a lot of rules it runs that can be problematic for some projects.
This can be overwritten in the spec at build time for some things by adding build steps that writes to `debian/rules`, e.g.:

```
- command: echo "override_dh_dwz:\n\t@" > debian/rules
```

This is at least a workable escape hatch but it would be nice if we could do better.
We could add something to the spec to do this but this doesn't seem particularly great since its specific to the underlying builder implementation.

Another idea I was thinking about is having an extra statement in our generated `debian/rules` that calls something like `-include dh_overrides.mk`.
This would allow users to include a file called `dh_overrides.mk` in their sources it will be automatically loaded into `debian/rules`.
If the file doesn't exist then make will ignore it.

**Comment by cpuguy83** [comment]
cc @adamperlin

**Comment by adamperlin** [comment]
I agree that adding something to the spec is not the best course of action here as it seems to break the spec abstraction of being distro nonspecific. I do like the idea of adding the `-include dh_overrides.mk` which will be included if a user specifically adds this as a source. We will just need to make sure this is well documented, though it will hopefully be a bit of an edge case.
